templates/cpp-template-default/proj.ios_mac/ios/RootViewController.mm

```diff
@@ -68,6 +69,13 @@ - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:animated];
 }
 
+//In iOS 12.0+, Screen Time's bug cause UIApplicationDidBecomeActiveNotification and UIApplicationWillResignActiveNotification do not fired
+//so we need to active CCDirectorCaller manually
+-(void)viewDidAppear:(BOOL)animated{
+    [super viewDidAppear:animated];
+    [[CCDirectorCaller sharedDirectorCaller] setActive:YES];
+}
```

Comment:
`CCDirectorCaller-ios.mm` already listen `UIApplicationWillEnterForegroundNotification` event, why need these codes too?

Reply:
> `CCDirectorCaller-ios.mm` already listen `UIApplicationWillEnterForegroundNotification` event, why need these codes too?

When the App starts up cold,UIApplicationWillEnterForegroundNotification would not be triggered . But the viewDidAppear would be called.

[https://developer.apple.com/documentation/uikit/uiapplicationwillenterforegroundnotification?language=objc](url)